### PR TITLE
Feature/lliss/basic structure

### PR DESCRIPTION
**Pull request by lliss**
This adds a basic database along with support for plugins.

You'll need to reprovision the vm since I recently merged the work that adds postgres.

### CODE REVIEW

File: defaults.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+dbUser: 'probodb'
+dbPassword: 'password'
+dbName: 'probodb'
+dbType: 'postgres'
+dbHost: 'localhost'
+eventGroup: 'proboDb'
+eventTopic: 'build_events'
+eventVersion: 1
+pluginDirectory: null
+knex: {}
```
Comment (tortillaj) [inline defaults.yaml]: maybe `nodeEnv: production` to make the [executable configurable](https://github.com/ProboCI/probo-db/pull/1/files#diff-ea4f1a242219c0bdf45c119a73a8fa20R16)

File: lib/Server.js [hunk 2]
```diff
@@ -1,35 +1,78 @@
...
-  }
-  storeBuildInDatabase(build, done) {
-    // Do stuff
-    let error = null;
-    done(null);
-  }
+    this.db = options.db;
+    this.config = options.config;
+
+    this.consumer = new eventbus.plugins['Kafka'].Consumer({
```
Comment (tortillaj) [inline lib/Server.js]: The other open source projects make the eventbus plugin configurable. We should do that here just for consistency.

See here:
https://github.com/ProboCI/probo/blob/master/defaults.yaml#L186
https://github.com/ProboCI/probo/blob/ef8148938e8c9ea37be12d1a1e8b4218a9f42cb6/lib/ContainerManager.js#L169

File: lib/Server.js [hunk 3]
```diff
@@ -1,35 +1,78 @@
...
+  }
+
+  testConnection(done) {
+    const self = this;
+    console.log('Attempting to connect to DB');
+
+    self.db.test()
+      .then(function(result) {
+        console.log(result.rows, 'RESULT');
+        console.log('Connection successful!');
```
Comment (tortillaj) [inline lib/Server.js]: Can logs be sent through bunyan and ultimately elasticsearch like the rest of our services?

File: defaults.yaml (hunk 1 above)
Comment (lliss) [inline defaults.yaml]: It is configurable now, you just have to start the server this way:
`NODE_ENV=development ./bin/probo-db`

File: lib/Server.js (hunk 2 above)
Comment (lliss) [inline lib/Server.js]: Makes sense.

File: lib/Server.js (hunk 3 above)
Comment (lliss) [inline lib/Server.js]: Sure. I'll set up a logger the way we do on the other projects.

File: defaults.yaml (hunk 1 above)
Comment (tortillaj) [inline defaults.yaml]: Good point :)

**Comment by lliss** [comment]
@tortillaj These most recent commits should address all of your comments.

**Comment by tortillaj** [comment]
end

launch it!!!!!!!!!!!!!!!!!!!!

**Pull request merged**